packages/react/src/ActionList/shared.ts

```diff
@@ -11,7 +11,7 @@ export type ActionListItemProps = {
    * Callback that will trigger both on click selection and keyboard selection.
    * This is not called for disabled or inactive items.
    */
-  onSelect?: (event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>) => void
+  onSelect?: (event: React.MouseEvent | React.KeyboardEvent) => void
```

Comment:
That is very true 😭, It's been something I've been dealing with in the [integration test PR](https://github.com/github/github/pull/315374). The main issue I had was, I couldn't find a decent way to express what element type `ActionList.Item` was based on the different conditions we have (i.e. passing a specific role to `ActionList` may cause `ActionList.Item` to be `HTMLLiElement` instead of `HTMLButtonElement`).

Reply:
I'm guessing there are TS errors with something like as well?

```
onSelect?: (event: React.MouseEvent<HTMLLIElement | HTMLButtonElement> | React.KeyboardEvent<HTMLLIElement | HTMLButtonElement>) => void
```

Reply:
Nope, I don't get any TS errors in PRC! This could be worth a try in the integration PR :eyes:. Would prefer this, especially if it doesn't require any additions/removals to existing types in dotcom.

Reply:
@siddharthkp sorry for ghosting you on the ts question!

>@broccolinisoup is there typescript magic that we can do to infer between button and list so that we can make the types: React.MouseEvent<HTMLLIElement | HTMLButtonElement> instead?
I actually remember we talked about maybe updating dotcom to use `ActionListItemProps['onSelect']`and then it infer the types. @TylerJDev should we give this a go?

Also could you share the integration PR with us and I can look at the logs and run it on the code space as well to have some more thoughts?

Reply:
Definitely, here's the integration PR: https://github.com/github/github/pull/317139. I'll add this in the description too.